src/boot/efi/boot.c

```diff
@@ -1278,6 +1302,54 @@ static void config_defaults_load_from_file(Config *config, char *content) {
                         }
+                } else if (streq8(key, "color-normal-foreground")) {
+                        uint8_t color;
+                        if(!parse_console_color(value, &color)) {
+                                log_error("Unknown color %s", value);
+                        }
+                        config->color_normal = (config->color_normal & 0xf0) | color;
+                } else if (streq8(key, "color-normal-background")) {
+                        uint8_t color;
+                        if(!parse_console_color(value, &color)) {
+                                log_error("Unknown color %s", value);
+                        }
+                        config->color_normal = (color << 4) | (config->color_normal & 0x0f);
+                } else if (streq8(key, "color-entry-foreground")) {
+                        uint8_t color;
+                        if(!parse_console_color(value, &color)) {
+                                log_error("Unknown color %s", value);
+                        }
+                        config->color_entry = (config->color_entry & 0xf0) | color;
+                } else if (streq8(key, "color-entry-background")) {
+                        uint8_t color;
+                        if(!parse_console_color(value, &color)) {
+                                log_error("Unknown color %s", value);
+                        }
+                        config->color_entry = (color << 4) | (config->color_entry & 0x0f);
+                } else if (streq8(key, "color-highlight-foreground")) {
+                        uint8_t color;
+                        if(!parse_console_color(value, &color)) {
+                                log_error("Unknown color %s", value);
+                        }
+                        config->color_highlight = (config->color_highlight & 0xf0) | color;
+                } else if (streq8(key, "color-highlight-background")) {
+                        uint8_t color;
+                        if(!parse_console_color(value, &color)) {
+                                log_error("Unknown color %s", value);
+                        }
+                        config->color_highlight = (color << 4) | (config->color_highlight & 0x0f);
+                } else if (streq8(key, "color-edit-foreground")) {
+                        uint8_t color;
+                        if(!parse_console_color(value, &color)) {
+                                log_error("Unknown color %s", value);
+                        }
+                        config->color_edit = (config->color_edit & 0xf0) | color;
+                } else if (streq8(key, "color-edit-background")) {
+                        uint8_t color;
+                        if(!parse_console_color(value, &color)) {
+                                log_error("Unknown color %s", value);
+                        }
+                        config->color_edit = (color << 4) | (config->color_edit & 0x0f);
                 }
```

Comment:
i really dislike the repetitiveness of this. please instead define a second string table with the color designations, i.e. a string table indexed by the earlier mentioned enum, that maps to strings "normal", "entry", "hightlight", "edit". Then write a generic handler help func that parses this generically. i.e. ooks for the "color-" prefix and the "-foreground"/"-background" suffix of the string. Then looks up the middle part in this 2nd string table, and the value in the 1st string table. and then fills in the array